modules/nf-core/seqtk/sample/main.nf

```diff
@@ -0,0 +1,57 @@
+process SEQTK_SAMPLE {
+    tag "${meta.id}"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/seqtk:1.4--he4a0461_1' :
+        'biocontainers/seqtk:1.4--he4a0461_1' }"
+
+    input:
+    tuple val(meta), path(reads), val(sample_size)
+
+    output:
+    tuple val(meta), path("*.fastq.gz"), emit: reads
+    path "versions.yml"                , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args   = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    if (!(args ==~ /.*-s[0-9]+.*/)) {
```

Comment:
I'd suggest including a preceding space in the pattern here:

```groovy
/.*\ -s[0-9]+.*/
```
... since otherwise some filename like `some-sequences-from-s3.fasta` included in the args would be picked up.

And there might also be a space before the number as well?, so perhaps we need:

```groovy
/.*\ -s(\ )?[0-9]+.*/
```

Reply:
I just copied the nf-core module! Do you reckon option 2 is best?

Reply:
Yes. Just double-checked with the nf console ...

The current one indeed does match filenames with `-s[0-9`, and does not match some which should be matched:
![image](https://github.com/user-attachments/assets/0648d237-8879-4622-b02c-a5fa14186067)

Number 2 does, however (I skipped the extra parens around the space which don't seem to be needed though):
![image](https://github.com/user-attachments/assets/101d6deb-8d14-41dc-b04b-80bd6b74ae97)

So, `/.*\ -s\ ?[0-9]+.*/` it is!

Reply:
@ryanjameskennedy But yea, I see what you mean, perhaps this should be reported upstream intead, hmm ...

Reply:
(For the record, I [sent a PR](https://github.com/nf-core/modules/pull/7867)).

Reply:
@ryanjameskennedy They merged my PR upstream, so you might want to pull again :)